### DISCUSSION TITLE
ModelCheckpoint with multiple validation dataloaders

### DISCUSSION THREAD

**Discussion by kkirtac** [Lightning-AI/pytorch-lightning#6664] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi all. How can one make ModelCheckpoint work with multiple val dataloaders? Currently, it receives a single monitor parameter. I would like to make the checkpointing happen only when both dataloaders show improvement. Should I subclass ModelCheckpoint to receive a monitor per dataloader?

**Comment by awaelchli** [comment, marked as answer]
You can compute the min or max (depending if your improvement is a decreasing or increasing metric) of both metrics and use self.log to log that combined value. Then, you only have one monitor for both and save_top_k can work on that.
